File: VNCore/Novel/Position.cs
using System;

namespace VNCore.Novel
{
    public struct Position
    {
        public sbyte X { get; set; }
        public sbyte Y { get; set; }
        public sbyte Width { get; set; }
        public sbyte Height { get; set; }
        public override string ToString()
        {
            return string.Format("{0} {1} {2} {3}", X, Y, Width, Height);
        }
        public static bool TryParse(string data, out Position result)
        {
            try
            {
                result = Parse(data);
                return true;
            }
            catch
            {
                result = new Position();
                return false;
            }
        }
        public static Position Parse(string data)
        {
            var dataArray = data.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
            var result = new Position();
            result.X = sbyte.Parse(dataArray[0]);
            result.Y = sbyte.Parse(dataArray[1]);
            result.Width = sbyte.Parse(dataArray[2]);
            result.Height = sbyte.Parse(dataArray[3]);
            return result;
        }
    }
}

File: VNCore.Java/Novel/Position.java
package Novel;

public class Position {
    public Byte x;
    public Byte y;
    public Byte width;
    public Byte height;

    @Override
    public String toString() {
        return String.format("%1d %2d %3d %4d", x, y, width, height);
    }

    public static Position parse(String data) {
        String[] dataArray = data.split(" ");
        Position result = new Position();
        result.x = Byte.parseByte(dataArray[0]);
        result.y = Byte.parseByte(dataArray[1]);
        result.width = Byte.parseByte(dataArray[2]);
        result.height = Byte.parseByte(dataArray[3]);
        return result;
    }

}